src/Search.php

```diff
@@ -5441,8 +5441,8 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
         if ($use_subquery_on_id_search || $use_subquery_on_text_search) {
             // Compute tables and fields names
             $main_table = getTableForItemType($itemtype);
-            $fk = getForeignKeyFieldForTable($main_table);
             $beforejoin = $searchopt[$ID]['joinparams']['beforejoin'];
+            $fk = $beforejoin['joinparams']['field'] ?? getForeignKeyFieldForTable($main_table);
```

Comment:
There may be cases where `$beforejoin[‘joinparams’][‘field’]` could be filled in, even though the **search option** uses the **sub_query** or the table in question is a relational table (`itemtype` / `items_id`). 

Would it be possible to condition `$fk` as follows to ensure that this fix is only applied in these specific conditions and to limit its scope? (The `Search` class is rather sensitive)  

```php
if ($use_subquery && $beforejoin['joinparams']['table'] === relation_table_with_items_id_and_itemtype) {
    $beforejoin['joinparams']['field'];
} else {
    $getForeignKeyFieldForTable($main_table);
}
```